src/grammar.js

```diff
@@ -43,7 +43,29 @@ class Grammar {
    * @return New Grammar object that represents
    */
   static fromNFA (nfa) {
-
+    const firstSymbol = nfa.start
+    const productions = {}
+    let aux
+    for (let state in nfa.table) {
+      aux = []
+      for (let terminalSymbol in nfa.table[state]) {
+        for (let nonTerminalSymbol of nfa.table[state][terminalSymbol]) {
+          aux.push(terminalSymbol + nonTerminalSymbol)
+          if (nfa.accept.has(nonTerminalSymbol) && aux.indexOf(nonTerminalSymbol) === -1) {
```

Comment:
If aux is a set, it will remove the necessity of the second test in the if. Because adding same thing to set will not change it.